auth: implemented login interface

doflamingo721 opened the pull request:
1. login api consumption,
2. storing token in SharedPreferences

doflamingo721 committed:
Major Project Structure Changes

Change Logs:
1. Project structure changes,
2. Modified color palette of home page,
3. Modified Login Page UI,
4, Added validations for zprn during login,
5, Added api and model directories in project structure,
6. Added new images in assets/images
7. Updated pubspec.yaml with new dependencies,
8.  Other minor changes

doflamingo721 committed:
auth: implemented login interface

drigio commented on lib/src/api/login_api_service.dart:
```diff
@@ -0,0 +1,29 @@
+class APIService {
+  Future<Map> login(String userId, String password) async {
+    dynamic postBody = <String, String>{"userId": userId, "password": password};
+    String url = "http://52.136.112.220:57842/api/auth/login";
```
Create a constant for `server address` url and then append it with your route. For eg here add constant `serverURL` to `libs/constants`. Import that constant here. then create a new constant `loginRoute` in this file, which has `serverURL+"/api/auth/login"`.

doflamingo721 replied:
Yes, I will do the needful in the next PR.

drigio commented on lib/src/api/login_api_service.dart:
```diff
@@ -0,0 +1,29 @@
+    dynamic postBody = <String, String>{"userId": userId, "password": password};
+    String url = "http://52.136.112.220:57842/api/auth/login";
+    var headers = {'Content-Type': 'application/json'};
+    SharedPreferences prefs = await SharedPreferences.getInstance();
```
Save the `access_token` into `FlutterSecureStorage`. And save `is_logged_in` = true in `SharedPrefs`.

drigio commented on lib/src/model/login_access.dart:
```diff
@@ -0,0 +1,32 @@
+// class LoginResponseModel {
```
Dont git stage this file if you arent supposed to upload it to the remote repo.

drigio commented on lib/src/pages/profile_page.dart:
```diff
@@ -0,0 +1,465 @@
+import 'package:flutter/material.dart';
+import 'package:final_app/themes.dart';
+
+// class ProfilePage extends StatefulWidget {
```
Remove unnecessary comments and again push the code.

drigio commented on lib/themes.dart:
```diff
@@ -0,0 +1,7 @@
+import 'package:flutter/material.dart';
+
+const String kAppName = "Grievance Portal";
```
Why is app name `string` constant stored in `themes.dart` file. Use `constants.dart` for the same.

doflamingo721 committed:
auth: implemented token validation

1. storing token in shared preferences and implemented checkLoginStatus to check if the token is present or null,
2. UI: changes in feeds, added new custom card.
3. UI: added a splash screen on app startup,
4. code cleanup;

doflamingo721 committed:
login: added loading screen

1. Added a loading screen, now if the user is already logged in, they won't see the login screen.
2. Code cleanup.
